src/ZkFold/Symbolic/Compiler/ArithmeticCircuit/Internal.hs

```diff
@@ -168,10 +203,11 @@ instance
 --    'WitnessField' is a root hash of a Merkle tree for a witness.
 toVar ::
   forall a i. (Finite a, Binary a, Binary (Rep i)) =>
-  Witness (Var i) a -> ByteString
+  Witness (Var a i) a -> ByteString
 toVar witness = runHash @(Just (Order a)) $ witness $ \case
-  InVar inV -> merkleHash inV
-  NewVar newV -> M newV
+  SysVar (InVar inV) -> merkleHash inV
+  SysVar (NewVar newV) -> M newV
+  ConstVar cV -> merkleHash cV
```

Comment:
^ Is there any chance that these witness variables could clash, like if `toByteString inV == toByteString cV`?

Reply:
Oh, actually there is a good chance

Reply:
We can add a "from constant" case to `Prec` and use it in `fromConstant` instance which we can call here

Reply:
Well, not a really probable case (we'd need input vars to be Zp which need same amount of bits to encode as `a`), but still would be great to distinguish between variables and constants

Reply:
I added a `Const` label to the `Prec` enum but I wonder if it should also have an `Input` label for input variables too 🤔

Reply:
I don't think so, the code looks clean to me and I don't think we'd encounter any collisions soon (not before sha256 collision is discovered ;))